Scripted/CIP_ParenchymaAnalysis/CIP_ParenchymaAnalysis.py

```diff
@@ -330,22 +330,58 @@ def onCTSelect(self, node):
                 slicer.app.layoutManager().sliceWidget(color).sliceLogic().GetSliceCompositeNode().SetBackgroundVolumeID('None')
 
     def onLabelSelect(self, node):
-        self.labelNode = node
-        self.applyButton.enabled = bool(self.CTNode)  # and bool(self.labelNode)
-        self.preProcessingWidget.enableFilteringFrame(bool(self.CTNode))
-        self.preProcessingWidget.enableLMFrame(bool(not self.labelNode))        
-        SlicerUtil.changeLabelmapOpacity(0.5)
-        if self.labelNode:
-            self.preProcessingWidget.filterApplication.setChecked(1)
-            self.preProcessingWidget.filterApplication.setEnabled(0)
-            for color in ['Red', 'Yellow', 'Green']:
-                slicer.app.layoutManager().sliceWidget(color).sliceLogic().GetSliceCompositeNode().SetLabelVolumeID(self.labelNode.GetID())
-        else:
-            self.preProcessingWidget.filterApplication.setChecked(0)
-            self.preProcessingWidget.filterApplication.setEnabled(1)
-            for color in ['Red', 'Yellow', 'Green']:
-                slicer.app.layoutManager().sliceWidget(color).sliceLogic().GetSliceCompositeNode().SetLabelVolumeID('None')
-
+        if not node: 
```

Comment:
The segmentation to labelmap conversion has to happen in the logic, because if you want to use this module from another module then you cannot use anything that is in the module widget class. You can report error message from the logic by raising an exception and catching it in the widget class.

Reply:
OK, logical, will be changed tomorrow.